rust/src/common.rs

```diff
@@ -121,3 +122,23 @@ pub fn to_hex(input: &[u8]) -> String {
         |b| vec![char::from(CHARS[(b >>  4) as usize]), char::from(CHARS[(b & 0xf) as usize])]
     ).flatten().collect();
 }
+
+fn print_hex(islice: &[u8], oslice: &mut [u8]) {
+    static CHARS: &'static [u8] = b"0123456789abcdef";
+
+    for i in 0..islice.len() {
+        oslice[2*i] = CHARS[(islice[i] >>  4) as usize];
+        oslice[2*i+1] = CHARS[(islice[i] & 0xf) as usize];
+    }
+    oslice[2*islice.len()] = 0;
```

Comment:
is this to essentially turn `oslice` into a nul terminated string? If the goal is to create a c-string I wonder if it would be better go all the way and do something that returns a CString? (cc @jasonish )

Reply:
a `jb_set_hex` could replace all but the `datasets.c` use it seems, and would leave things in Rust

Reply:
Doing `jb_set_hex` in next version.
There is also `OutputFilestoreFinalizeFiles` which uses `PrintHexString`...

Reply:
This function looks like its solely designed to be used from C. So I think this makes sense as the result is being placed into a provided buffer.

If we were to build a CString, Rust would need to allocate and return it.  Then a call would have to be made back to Rust to free it.

I wonder if this should just be one function all within the `extern` function, to make it more clear that its really for C callers only.